pkg/virt-controller/watch/export/export.go

```diff
@@ -384,19 +393,143 @@ func (ctrl *VMExportController) handleIsSourcePvc(vmExport *exportv1.VirtualMach
 			}
 		}
 	} else {
-		if pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed {
-			// The server died or completed, delete the pod.
-			ctrl.Recorder.Eventf(vmExport, corev1.EventTypeWarning, exporterPodFailedOrCompletedEvent, "Exporter pod %s/%s succeeded or failed", pod.Namespace, pod.Name)
-			if err := ctrl.Client.CoreV1().Pods(vmExport.Namespace).Delete(context.Background(), pod.Name, metav1.DeleteOptions{}); err != nil {
+		if err := ctrl.handlePodSucceededOrFailed(vmExport, pod); err != nil {
+			return retry, err
+		}
+	}
+
+	return ctrl.updateVMExportPvcStatus(vmExport, pvcs, pod, service, inUse)
+
+}
+
+func (ctrl *VMExportController) handleIsVmSnapshot(vmExport *exportv1.VirtualMachineExport, service *corev1.Service) (time.Duration, error) {
+	var retry time.Duration
+	pvcs := make([]*corev1.PersistentVolumeClaim, 0)
```

Comment:
nit: This is weird syntax to me, but you seem to like it, so why is a zero length slice desirable, especially since it is reassigned before being referenced?  And if for some reason it is, the following declarations seem more standard to me

```
var pvcs = make([]*corev1.PersistentVolumeClaim, 0)
pvcs := []*corev1.PersistentVolumeClaim{}
```

To me, `var pvcs []*corev1.PersistentVolumeClaim` makes the most sense

Reply:
I don't know what I was thinking when I wrote that. Fixed